homeassistant/components/device_tracker/__init__.py

```diff
@@ -95,30 +103,43 @@ def setup(hass, config):
     """Setup device tracker."""
     yaml_path = hass.config.path(YAML_DEVICES)
 
-    conf = config.get(DOMAIN, {})
+    conf = CONFIG_SCHEMA(config).get(DOMAIN, [])
 
-    # Config can be an empty list. In that case, substitute a dict
-    if isinstance(conf, list):
-        conf = conf[0] if len(conf) > 0 else {}
+    def _get_conf_value(key_name, default):
+        """Retrieve a value from all platforms & rasie error if conflictind."""
+        res = list(set([plat.get(key_name) for plat in conf
+                        if plat.get(key_name) is not None]))
+        if len(res) > 1:
+            _LOGGER.warning('device_tracker: Conflicting %s settings: %s',
+                            key_name, ','.join(res))
+        if len(res) > 0:
+            return res[0]
+        return default
 
+    _LOGGER.warning("CONFIG: from device_tracker.setup() %s", conf)
     consider_home = timedelta(
-        seconds=util.convert(conf.get(CONF_CONSIDER_HOME), int,
-                             DEFAULT_CONSIDER_HOME))
-    track_new = util.convert(conf.get(CONF_TRACK_NEW), bool,
-                             DEFAULT_CONF_TRACK_NEW)
-    home_range = util.convert(conf.get(CONF_HOME_RANGE), int,
-                              DEFAULT_HOME_RANGE)
+        seconds=_get_conf_value(CONF_CONSIDER_HOME, DEFAULT_CONSIDER_HOME))
+    track_new = _get_conf_value(CONF_TRACK_NEW, DEFAULT_TRACK_NEW)
+
+    devices = load_config(yaml_path, hass, consider_home)
 
-    devices = load_config(yaml_path, hass, consider_home, home_range)
-    tracker = DeviceTracker(hass, consider_home, track_new, home_range,
-                            devices)
+    tracker = DeviceTracker(hass, consider_home, track_new, devices)
 
     def setup_platform(p_type, p_config, disc_info=None):
         """Setup a device tracker platform."""
         platform = prepare_setup_platform(hass, config, DOMAIN, p_type)
         if platform is None:
             return
 
+        # Apply component & platform specific validation schema
```

Comment:
Exactly what I thought, but it seem like after I added the CONFIG_SCHEMA to device_tracker, this is not happening any more [related to this elif?](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/bootstrap.py#L110)